lib/ex_unit/lib/ex_unit/doc_test.ex

```diff
@@ -303,14 +303,30 @@ defmodule ExUnit.DocTest do
           :ok
 
         actual ->
-          expr = "#{unquote(String.trim(expr))} === #{unquote(String.trim(expected))}"
-          error = [message: "Doctest failed", expr: expr, left: actual, right: expected]
+          doctest = unquote("\n" <> formatted <> "\n" <> expected)
+
+          last_exp =
+            unquote(expr)
+            |> String.split("\n")
```

Comment:
Why are we doing the `String.split` when we didn't do it before? I am afraid this can have false positives, when for example the last expression has new lines in it. Isn't it enough too do:

    expr = "#{unquote(String.trim(expr))} === #{unquote(String.trim(expected))}"

Like we did before?

Reply:
The original doctest would fail like this, following your example from above it looked like you wanted to omit everything else than the last line, since it will be present in the doctest key. 
```elixir
Doctest failed
     code:  a = "This is an egregiously long text string."
             b = ~r{an egregiously long}
             c = \"a slightly shorter\"
             String.replace(a, b, c) === "This is a much shorter text string."
     left:  "This is a slightly shorter text string."
     right: "This is a much shorter text string."
```
I agree the `String.split` is not the most pragmatic way and also why I ask for help about it